dlib/python/numpy_image.h

```diff
@@ -70,11 +73,9 @@ class numpy_rgb_image
     numpy_rgb_image (boost::python::object& img) 
     {
         long shape[3];
-        get_numpy_ndarray_parts(img, _data, shape);
+        get_numpy_ndarray_parts(img, _data, _contig_buf, shape);
         _nr = shape[0];
         _nc = shape[1];
-        if (shape[2] != 3)
```

Comment:
Ah, so it does.  Still, the code is a little hard to follow since reasoning
about its correctness requires pulling together these different little
details from various parts of the code.  It's better if the bodies of
functions are self evidently correct just own their own.  This isn't always
possible, but adding a little more to the if statements in some of these
functions can make them obviously correct independent of the specific error
handling happening in other functions.  Also, it is likely that other
people will edit these functions at some point and they will probably not
consider their effects on code that calls them.  For example, if the error
handling behavior of get_numpy_ndarray_shape() were to change then the
error handling behavior of the function we are talking about could easily
become incorrect.